File: WebClient SDK/src/default/workflow/krPermissions/krKeepReadCardPermissionStoreExtension.ts
import { ICardStoreExtensionContext, CardStoreExtension } from 'tessa/cards/extensions';
import { UIContext } from 'tessa/ui';
import { ICardEditorModel, ICardModel, CardEditorOperationType } from 'tessa/ui/cards';
import { KrToken } from 'tessa/workflow';
import { Guid } from 'tessa/platform';

export class KrKeepReadCardPermissionStoreExtension extends CardStoreExtension {
  public afterRequest(context: ICardStoreExtensionContext): void {
    const uiContext = UIContext.current;
    let editor!: ICardEditorModel;
    let model!: ICardModel;
    if (
      (editor = uiContext.cardEditor!) &&
      (model = editor.cardModel!) &&
      // Флаг сохраняем только при открытии при сохранении карточки, рефреш карточки скидывает режим редактирования
      (editor.currentOperationType === CardEditorOperationType.SaveAndRefresh ||
        editor.currentOperationType === CardEditorOperationType.Open)
    ) {
      let token: KrToken | null = null;
      if (context.response) {
        token = KrToken.tryGet(context.response.info);
      }

      if (
        token &&
        (Guid.equals(token.cardId, model.card.id) || Guid.equals(token.cardId, Guid.empty))
      ) {
        token.setInfo(context.request.info);
      }
    }
  }
}

File: Source/Extensions/Tessa.Extensions.Default.Client/Workflow/KrPermissions/KrKeepReadCardPermissionStoreExtension.cs
using System;
using System.Threading.Tasks;
using Tessa.Cards.Extensions;
using Tessa.Extensions.Default.Shared.Workflow.KrProcess;
using Tessa.UI;
using Tessa.UI.Cards;

namespace Tessa.Extensions.Default.Client.Workflow.KrPermissions
{
    public sealed class KrKeepReadCardPermissionStoreExtension : CardStoreExtension
    {
        public override Task AfterRequest(ICardStoreExtensionContext context)
        {
            IUIContext uicontext = UIContext.Current;

            ICardEditorModel editor;
            ICardModel model;
            if ((editor = uicontext.CardEditor) != null
                && (model = editor.CardModel) != null
                && (editor.CurrentOperationType == CardEditorOperationType.SaveAndRefresh
                || editor.CurrentOperationType == CardEditorOperationType.Open))
            {
                KrToken token = KrToken.TryGet(context.Response.Info);
                if (token != null && (token.CardID == model.Card.ID || token.CardID == Guid.Empty))
                {
                    token.Set(model.Card.Info);
                }
            }

            return Task.CompletedTask;
        }
    }
}
